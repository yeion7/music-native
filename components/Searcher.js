/* @flow weak */

import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { Header, Item, Input, Icon } from "native-base";

const Searcher = ({ handleChange, state }) => (
  <Header searchBar rounded>
    <Item>
      <Icon name="ios-search" />
      <Input
        autoCorrect={false}
        placeholder="Search"
        value={state.text}
        onChangeText={handleChange}
      />
    </Item>
  </Header>
);

export default Searcher;
