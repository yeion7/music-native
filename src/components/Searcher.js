/* @flow weak */

import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { Header, Item, Input, Icon } from "native-base";
import { connect } from "react-redux";
import { changeSearch } from "../actions/search";

const Searcher = ({ text, onChange }) => {
  return (
    <Header searchBar rounded>
      <Item>
        <Icon name="ios-search" />
        <Input
          autoCorrect={false}
          placeholder="Search"
          value={text}
          onChangeText={onChange}
        />
      </Item>
    </Header>
  );
};

function mapStateToProps(state, props) {
  console.log("estado", state), console.log("ownprops", props);
  return {
    text: state.search.text
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onChange: () => dispatch(changeSearch)
  };
}
export default connect(mapStateToProps, mapDispatchToProps)(Searcher);
