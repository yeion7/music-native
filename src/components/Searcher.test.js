import React from "react";
import "react-native";

import Searcher from "./Searcher";

import renderer from "react-test-renderer";

describe("renders test ", () => {
  it("render Seacher component", () => {
    const tree = renderer.create(<Searcher onChange={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
