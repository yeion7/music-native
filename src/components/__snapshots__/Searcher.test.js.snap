// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test  render Seacher component 1`] = `
<View>
  <View
    rounded={true}
    searchBar={true}
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomColor": "#a7a6ab",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 15,
        "right": 0,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "top": 0,
      }
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#CECDD2",
          "borderBottomWidth": 1,
          "borderColor": "transparent",
          "borderLeftWidth": 0,
          "borderRadius": 25,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flex": 1,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "flex-start",
          "marginLeft": 2,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <TextInput
        autoCorrect={false}
        editable={true}
        onChangeText={[Function]}
        placeholder="Search"
        placeholderTextColor="#575757"
        style={
          Object {
            "alignSelf": "center",
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 30,
            "lineHeight": 24,
            "paddingLeft": 5,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        virtual={undefined}
      />
    </View>
  </View>
</View>
`;
